File: mipush_hook/src/main/java/com/nihility/Hooked.java
package com.nihility;

import java.util.HashSet;
import java.util.Set;

public class Hooked {
    private final static Set<String> hookedRecord = new HashSet<>();

    public static boolean contains(String id) {
        return hookedRecord.contains(id);
    }

    public static void mark(String id) {
        hookedRecord.add(id);
    }
}

File: mipush_hook/src/main/java/com/nihility/Hooked.kt
package com.nihility

object Hooked {
    private val hookedRecord = hashSetOf<String>()

    @JvmStatic
    fun contains(id: String): Boolean = hookedRecord.contains(id)

    @JvmStatic
    fun mark(id: String) {
        hookedRecord.add(id)
    }
}
